crates/stages/src/stages/execution.rs

```diff
@@ -230,13 +232,16 @@ impl<DB: Database> Stage<DB> for ExecutionStage {
                             storage_id.clone(),
                             StorageEntry { key: hkey, value: old_value },
                         )?;
-
-                        if new_value.is_zero() {
-                            tx.delete::<tables::PlainStorageState>(
-                                address,
-                                Some(StorageEntry { key: hkey, value: old_value }),
-                            )?;
-                        } else {
+                        tracing::debug!(
+                            "{address} setting storage:{key} ({old_value} -> {new_value})"
+                        );
+
+                        // Always delete old value as duplicate table put will not override it
+                        tx.delete::<tables::PlainStorageState>(
```

Comment:
Is there no way to avoid two calls (delete then write) with update?

Reply:
No, this was a bug that I had, putting values with same `SubKey` will not replace that value but just add an additional one.